.github/workflows/check-demo-notebooks.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will check if the run.py files in every demo match the run.ipynb notebooks.
+
+name: Check Demos Match Notebooks
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+        - 'demo/**'
+
+jobs:
+  check-notebooks:
+    name: Generate Notebooks and Check 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Generate Notebooks
+        run: |
+          pip install nbformat && cd demo && python generate_notebooks.py
+      - name: Print Git Status
+        run: echo $(git status) && echo $(git diff)
+      - name: Assert Notebooks Match 
+        id: assertNotebooksMatch
+        run: git status | grep "nothing to commit, working tree clean"
+      - name: Comment On PR
```

Comment:
Does this run on forked repos?

Reply:
Good question, I just checked on this PR https://github.com/gradio-app/gradio/pull/2691 

The check fails as expected, but it [can't comment](https://github.com/gradio-app/gradio/actions/runs/3518591252/jobs/5897671236). Do you know how to fix that? Apparently it's an [open issue](https://github.com/actions/first-interaction/issues/10).

Reply:
We need to follow this approach I think: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/#:~:text=The%20main%20differences%20between%20the,but%20not%20from%20external%20forks. I think it may also be possible to comment on the pr if the action is defined in a separate repo? I think that's what doc-builder does.

@pngwn May now some best practices for commenting on PRs from forks.

Reply:
Ah nice